pkg/clientcache/cache.go

```diff
@@ -85,24 +85,24 @@ func (c *ClientCache) CreateClient(contextName string, restConfig *rest.Config)
 }
 
 // CreateRemoteClientsFromSecret is a convenience method for testing purposes
-func (c *ClientCache) CreateRemoteClientsFromSecret(namespacedName types.NamespacedName) error {
+func (c *ClientCache) CreateRemoteClientsFromSecret(secretKey types.NamespacedName) error {
 	if c.localClient == nil {
 		return errors.New("creating from secret requires local client to be set")
 	}
 
-	apiConfig, err := c.extractClientCmdFromSecret(namespacedName)
+	apiConfig, err := c.extractClientCmdApiConfigFromSecret(secretKey)
 	if err != nil {
 		return err
 	}
 
-	for _, ctx := range apiConfig.Contexts {
-		clientCfg := clientcmd.NewNonInteractiveClientConfig(*apiConfig, ctx.Cluster, &clientcmd.ConfigOverrides{}, nil)
-		cfg, err := clientCfg.ClientConfig()
+	for ctx, _ := range apiConfig.Contexts {
```

Comment:
I think it's more correct to use the context name and not the cluster name, since that's what `NewNonInteractiveClientConfig` expects to receive. But I might be wrong.

Reply:
Good catch. I think we got lucky because the context names and cluster names that we use in tests are the same.